neo/Consensus/ConsensusService.cs

```diff
@@ -239,8 +277,55 @@ private void ExtendTimerByFactor(int maxDelayInBlockTimes)
                 ChangeTimer(nextDelay);
         }
 
+        private void TryToSaveFuturePayloads(ConsensusPayload payload)
+        {         
+            if (payload.BlockIndex > context.Block.Index)
```

Comment:
```c#
if (payload.BlockIndex == context.Block.Index + 1)
```

What about this?

Reply:
That's what I meant about `H+1`, brother @vncoelho ... :joy: not H+1 payloads, but payloads with height == height + 1, like Erik said.

Reply:
Nooo, this is not good. We are still going to lose payloads.

We need to storage all future ones because we can easily persist 1-5 blocks in some few couple of seconds of `ms`.

What we can do it to limit this:

`if (payload.BlockIndex >= context.Block.Index + 1 && payload.BlockIndex <= context.Block.Index + 10)`

Reply:
Check https://github.com/neo-project/neo/pull/949/commits/8411857b6bbf95c08d1b26809b412e761f96a735

Reply:
@erikzhang, did you see the limits? I think that this is enough, right?

As we discussed, just store the next might not be the most efficient. By having a ranged limit of 10-100 blocks we are going to be sure that this mechanism may help.

Because sometimes blocks can be synced quite quickly in a batch.